02-oop-inheritance/script.js

```diff
@@ -0,0 +1,73 @@
+// 1 DONE- 4 - 8 DONE
+function Movie(title, year, duration) {
+	this.title = title;
+	this.year = year;
+	this.duration = duration;
+	this.cast = new Array();
+	this.play = function() {};
+	this.pause = function(){};
+	this.resume = function(){}
+	this.addCast = function(actors) {
+		if(Array.isArray(actors)) {
+			for(let i = 0; i<actors.length; i++) {
+				this.cast.push(actors[i]);
+			}
+		} else {
+			this.cast.push(actors);
+		}
+
+	};
+	this.showCast = function() {
+		for(let i = 0; i<this.cast.length; i++) {
+			console.log(this.cast[i]);
+		}
+	}
+}
+
+// 2 DONE
+let movie1 = new Movie("Notebook", 2004, 123);
+let movie2 = new Movie("Star Wars: Episode II - Attack of the Clones", 2002, 142);
+let movie3 = new Movie("SlumdogMillonaire", 2008, 120);
+
+// 3
+function EventEmitter() {
```

Comment:
You can store an inner state that is handled with those methods:

```
this.eventHandlers = [
 { 
   name: 'play', 
   callback: function() { 
     console.log('played') 
  },
 { 
   name: 'stop', 
   callback: function() { 
     console.log('stopped') 
  },
 }
]
```

On: allows us to add handler objects to the list. 
Emit: allows us to execute the specified event callback if exists.
Off: allows us to remove the handler object from the list if exists.